flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Utils/arinc429.js

```diff
@@ -0,0 +1,43 @@
+class Arinc429Word {
+    constructor(value) {
+        Arinc429Word.f64View[0] = value;
+
+        this.ssm = Arinc429Word.u32View[0];
+        this.value = Arinc429Word.f32View[1];
+    }
+
+    static empty() {
+        return new Arinc429Word(0);
+    }
+
+    static fromSimVarValue(name) {
+        return new Arinc429Word(SimVar.GetSimVarValue(name, "number"));
+    }
+
+    isNormal() {
+        return this.ssm === Arinc429Word.SignStatusMatrix.NormalOperation;
+    }
+
+    /**
+     * Returns the value when normal, the supplied default value otherwise.
+     */
+    valueOr(defaultValue) {
+        return this.isNormal() ? this.value : defaultValue;
+    }
+
+    equals(other) {
+        return this === other
+            || (typeof other !== "undefined" && other !== null && this.value === other.value && this.ssm === other.ssm);
+    }
+}
+
+Arinc429Word.f64View = new Float64Array(1);
+Arinc429Word.u32View = new Uint32Array(Arinc429Word.f64View.buffer);
+Arinc429Word.f32View = new Float32Array(Arinc429Word.f64View.buffer);
+
+Arinc429Word.SignStatusMatrix = Object.freeze({
+    FailureWarning: 0,
+    FunctionalTest: 1,
+    NoComputedData: 2,
```

Comment:
One element that surprised me here is the order, encoding Functional Test as 10 instead of 01. ARINC has an interesting quirk where it's usually specified back to front (last bit first) unlike other serial formats, so the matrix I'm more familiar with looks like this:

| Bit 31 | Bit 30 | Meaning         |
|--------|--------|-----------------|
| 0      | 0      | Normal          |
| 1      | 0      | NCD             |
| 0      | 1      | Functional Test |
| 1      | 1      | Failure Warning |

This order also corresponds more closely to the status priority. If multiple statuses apply, the sending system should choose FW first, then FT, then NCD, and then finally Normal only if nothing else applies.

While I don't have a concrete suggestion here, I know little about how WebAssembly chooses it's endianness. All in all, maybe this quirky divergence from the ARINC spec warrants an extra comment, that it's due to a flipped bit order compared to the docs.

Reply:
I referred to the [Wikipedia page](https://en.wikipedia.org/wiki/ARINC_429) on ARINC 429, where I believe I took the representation for BNR encoding instead of the representation for discrete data. All in all I have no opinion on which of those to pick, and thus will change it into the version you suggest as by all means I consider you to be the ARINC 429 specification master 😀.